bi_sql_editor/models/bi_sql_view.py

```diff
@@ -0,0 +1,568 @@
+        for sql_view in self:
+            sql_view.materialized_text =\
+                sql_view.is_materialized and 'MATERIALIZED' or ''
+
+    @api.depends('technical_name')
+    @api.multi
+    def _compute_view_name(self):
+        for sql_view in self:
+            sql_view.view_name = '%s%s' % (
+                sql_view._sql_prefix, sql_view.technical_name)
+
+    @api.depends('technical_name')
+    @api.multi
+    def _compute_model_name(self):
+        for sql_view in self:
+            sql_view.model_name = '%s%s' % (
+                sql_view._model_prefix, sql_view.technical_name)
+
+    @api.onchange('group_ids')
+    def onchange_group_ids(self):
+        if self.state not in ('draft', 'sql_valid'):
+            self.has_group_changed = True
+
+    # Overload Section
+    @api.multi
+    def unlink(self):
+        non_draft_views = self.search([
+            ('id', 'in', self.ids),
+            ('state', 'not in', ('draft', 'sql_valid'))])
+        if non_draft_views:
+            raise UserError(_("You can only unlink draft views"))
+        return super(BiSQLView, self).unlink()
+
+    @api.multi
+    def copy(self, default=None):
+        self.ensure_one()
+        default = dict(default or {})
+        default.update({
+            'name': _('%s (Copy)') % (self.name),
+            'technical_name': '%s_copy' % (self.technical_name),
+        })
+        return super(BiSQLView, self).copy(default=default)
+
+    # Action Section
+    @api.multi
+    def button_create_sql_view_and_model(self):
+        for sql_view in self:
+            if sql_view.state != 'sql_valid':
+                raise UserError(_(
+                    "You can only process this action on SQL Valid items"))
+            # Create ORM and acess
+            sql_view._create_model_and_fields()
+            sql_view._create_model_access()
+
+            # Create SQL View and indexes
+            sql_view._create_view()
+            sql_view._create_index()
+
+            if sql_view.is_materialized:
+                sql_view.cron_id = self.env['ir.cron'].create(
+                    sql_view._prepare_cron()).id
+            sql_view.state = 'model_valid'
+
+    @api.multi
+    def button_set_draft(self):
+        for sql_view in self:
+            if sql_view.state in ('model_valid', 'ui_valid'):
+                # Drop SQL View (and indexes by cascade)
+                sql_view._drop_view()
+
+                # Drop ORM
+                sql_view._drop_model_and_fields()
+
+            sql_view.tree_view_id.unlink()
+            sql_view.graph_view_id.unlink()
+            sql_view.pivot_view_id.unlink()
+            sql_view.search_view_id.unlink()
+            sql_view.action_id.unlink()
+            sql_view.menu_id.unlink()
+            sql_view.rule_id.unlink()
+            if sql_view.cron_id:
+                sql_view.cron_id.unlink()
+            sql_view.write({'state': 'draft', 'has_group_changed': False})
```

Comment:
This is what happens when running the test several times. The records are never deleted. It seems I need to run some queries every time I run the tests:
![image](https://user-images.githubusercontent.com/19620251/28662621-10cfb594-72bb-11e7-9c10-8f481c5858cd.png)

Moreover this have happened after running the uninstall hook, which looks like:
` ```
   cr.execute("""
        select relname
        from pg_class
        where relname like 'x_bi_sql_view%' and relkind='m'
        """)
    for r in cr.fetchall():
        cr.execute("""
               DROP MATERIALIZED VIEW %s
            """ % r)
```

    cr.execute("""
        select relname
        from pg_class
        where relname like 'x_bi_sql_view%' and relkind='r'
        """)
    for r in cr.fetchall():
        cr.execute("""
               DROP TABLE %s
            """ % r)
    cr.execute("""
        select table_name from INFORMATION_SCHEMA.views
        where table_name like 'x_bi_sql%'""")

    # Drop not materialized views
    for v in cr.fetchall():
        cr.execute("""
               DROP VIEW %s
            """ % v)

    #Drop table if uninstalling went wrong
    cr.execute("""
        delete from ir_model_fields where model like 'bi.sql.view%';
        delete from ir_model_fields where model like 'bi_sql_%';
        delete from ir_model where model like 'x_bi_sql_view.%';
        """)`